programs/bpf_loader/src/lib.rs

```diff
@@ -226,8 +227,13 @@ pub fn create_executor_from_account(
     };
 
     if let Some(ref tx_executor_cache) = tx_executor_cache {
-        if let Some(executor) = tx_executor_cache.get(program.get_key()) {
-            return Ok((executor, None));
+        match tx_executor_cache.get(program.get_key()) {
+            // Executor exists and is cached, use it
+            Some(Some(executor)) => return Ok((executor, None)),
+            // We cached that the Executor does not exist, abort
+            Some(None) => return Err(InstructionError::InvalidAccountData),
```

Comment:
This looks a little scary since it is a new error conditions that is not directly featurized.  Afaict this case can only occur once the `delay` feature is activated but that is a bit distant and could potentially lead to confusion or an missed code path

Reply:
Hm yes, I added a comment to point out that this case only exists when the feature is active.
The thing is, if we made a condition here that would check the feature, it would still result in an error being returned.
Not much gained IMO.

Reply:
Yeah, comment would help